PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetExtractor.cxx

```diff
@@ -1163,71 +1140,65 @@ void AliAnalysisTaskJetExtractor::GetTrueJetPtFraction(AliEmcalJet* jet, Double_
-{
-  // need to specify which array name to pick based on if this is the second matching
-  TString jet_array_name = "";
-  if(isPartLevelMatching)
-    jet_array_name = fMatchedPartLevelJetsArrayName;
-  else
-    jet_array_name = fMatchedDetLevelJetsArrayName;
-
-  // #################################################################################
-  // ##### OBSERVABLES FROM MATCHED JETS: Jet pt, jet mass + angularity + pTD(momentum dispersion)
-  matchedJetDistance = 8.; // 8 is higher than maximum possible matching distance
-  if(jet_array_name != "")
-  {
-    // only look at background for first Matching (don't need this for truth level-info)
-    AliRhoParameter* rho = 0;
-    AliRhoParameter* rhoMass = 0;
-    Double_t trueRho = 0;
-    if(!isPartLevelMatching){
-      // "True" background for pt
-      rho = static_cast<AliRhoParameter*>(InputEvent()->FindListObject(fMatchedDetLevelJetsRhoName.Data()));
-      if(rho)
-       trueRho = rho->GetVal();
-      // "True" background for mass
-      rhoMass = static_cast<AliRhoParameter*>(InputEvent()->FindListObject(fMatchedDetLevelJetsRhoMassName.Data()));
-    }
-    TClonesArray* matchedJetArray = static_cast<TClonesArray*>(InputEvent()->FindListObject(Form("%s", jet_array_name.Data())));
-
-    // Loop over all true jets to find the best match & fill properties
-    matchedJetPt = 0;
-    matchedJetMass = 0;
-    if(matchedJetArray)
-    {
-      AliEmcalJet* bestMatchedJet = 0;
-      for(Int_t i=0; i<matchedJetArray->GetEntriesFast(); i++)
-      {
-        AliEmcalJet* matchedJet = static_cast<AliEmcalJet*>(matchedJetArray->At(i));
-        if(matchedJet->Pt() < 0.15)
-          continue;
-        Double_t deltaR = GetDistance(matchedJet->Eta(), jet->Eta(), matchedJet->Phi(), jet->Phi());
-        // Search for the best match
-        if(deltaR < matchedJetDistance)
-        {
-          matchedJetDistance = deltaR;
-          bestMatchedJet =  matchedJet;
-        }
-      }
-      // Fill properties for best match
-      if(bestMatchedJet)
-      {
-        matchedJetPt   = bestMatchedJet->Pt() - bestMatchedJet->Area()* trueRho;
-        if(rhoMass)
-          matchedJetMass = bestMatchedJet->GetShapeProperties()->GetSecondOrderSubtracted();
-        else
-          matchedJetMass = bestMatchedJet->M();
-
-        Double_t leSub_noCorr = 0; Double_t trackPtMean = 0; Double_t trackPtMedian = 0;
-        matchedJetAngularity = 0; matchedJetpTD = 0;
-        CalculateJetShapes(bestMatchedJet, leSub_noCorr, matchedJetAngularity, matchedJetpTD, trackPtMean, trackPtMedian);
-      }
-    }
+void AliAnalysisTaskJetExtractor::DoJetMatching(){
+   // Perform the matching before the main jet loop (Only if we use the tagger for matching)                                              
```

Comment:
For completeness, this comment should maybe be adapted (we always use our internal tagger here).

Reply:
Yup sounds good!